File: scripts/algorithms/Sum of Left Leaves/Sum of Left Leaves.java
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        
        if(root==null)
            return 0;
        
        if(root.left!=null && root.left.left==null && root.left.right==null){
            return root.left.val + sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
        }else{
            return sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
        }
        
        
    }
}

File: scripts/algorithms/S/Sum of Left Leaves/Sum of Left Leaves.py
def is_leaf(x):
    return x.left is None and x.right is None

class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        if root.left and is_leaf(root.left):
            left = root.left.val
        else:
            left = self.sumOfLeftLeaves(root.left)
        return left + self.sumOfLeftLeaves(root.right)
